### PR TITLE
Add duplicate name restriction to 'Rename Category' command (fix #140)

### LINKED ISSUE #140: Add duplicate name restriction to 'Rename Category' command
Currently 'Rename Category' is not restricted about duplicate names as the "New Category" command is.

### PR DESCRIPTION
**Pull request by Gasparoken**
fix #140

### CODE REVIEW

**Comment by dacap** [comment]
I've tested a little this one and I'd change that if we specify the same name for the category, nothing is done (no transaction, no error message, we've just put the same name to the category).

**Comment by Gasparoken** [comment]
Ok (IMO, it is better that the use of a duplicate name be flagged during the process of creating or changing a category's name. Just to avoid duplicate names in the table of categories)

**Comment by Gasparoken** [comment]
Ready for review.

**Comment by dacap** [comment]
I don't know if what I said was understood, but basically the code should be the same but if the tileset/cateogry that is found is exactly the same as the one that we're renaming, we just do nothing (instead of showing an error message as the original impl).

File: main.lua [hunk 1]
```diff
@@ -1388,34 +1395,34 @@ local function new_or_rename_category_dialog(categoryTileset)
...
-      end)
-    end
+  if not data.ok then
+    return
+  elseif data.name == "" then
+    return app.alert("Empty names are not allowed.")
+  -- Check that we cannot create two tilesets with the same name
+  elseif find_tileset_by_name(spr, data.name) then
+    return
+  end
```
Comment (dacap) [inline main.lua:1407]: Please go back to the first implementation, just do `return` if `find_tileset_by_name` returns `categoryTileset`.

Comment (Gasparoken) [reply main.lua:1407]: I think this is already done (a few minutes before your comment)

Comment (dacap) [reply main.lua:1407]: We need something like this:

```lua
  if not data.ok then
    return
  elseif data.name == "" then
    return app.alert("Empty names are not allowed.")
  end

  -- Check that we cannot create two tilesets with the same name
  local ts = find_tileset_by_name(spr, data.name)
  if ts then
    if ts == categoryTileset then
      return
    end
    return app.alert("A category named '" .. data.name .. "' already exist. " ..
                     "You cannot have two categories with the same name")
  end

  if categoryTileset then
    ...
```

**Comment by Gasparoken** [comment]
Sorry for my confusion. Ready for merge.

**Pull request merged**